### PR TITLE
Add notarization

### PR DESCRIPTION
**Pull request by natewalck**
This adds notarization functions and calls to them.

For this to work in the Github action, we need two new bits of information set as an env var:

```
DEV_ACCOUNT
NOTARY_PASS
```
Where DEV_ACCOUNT is the account the notary passcode has come from and NOTARY_PASS is the notary passcode itself.

### CODE REVIEW

**Comment by natewalck** [comment]
Added the repo secrets, let's see if this works :D

**Comment by natewalck** [comment]
This couldn't sign, so the notarization wasn't tested. We may have to YOLO to test that part :(

**Comment by erikng** [comment]
Why don't you steal the pattern from nudge?

**Comment by natewalckcdo** [comment]
I forgot nudge did this :D I recently had to do this for another project, so I was familiar with this one.

**Comment by natewalck** [comment]
I forgot nudge had this, actually :D I recently had to do this for another project, so I was familiar with this pattern.

**Comment by natewalck** [comment]
Will `--wait` *actually* wait for it to be done? If so, that pattern is much nicer.

**Comment by natewalck** [comment]
Added the nudge method, hoping I got all the bits right. I *think* we can get away with just notarizing the package due to the nature of the contents.

File: build_python_framework_pkgs.zsh [hunk 1]
```diff
@@ -18,6 +18,10 @@ RP_BINDIR="/tmp/relocatable-python"
 MP_BINDIR="/tmp/munki-pkg"
 CONSOLEUSER=$(/usr/bin/stat -f "%Su" /dev/console)
 PIPCACHEDIR="/Users/${CONSOLEUSER}/Library/Caches/pip"
+# NOTARY_PASS="" # Store as a repo secret
+XCODE_PATH="/Applications/Xcode_13.2.1.app"
```
Comment (natewalck) [inline build_python_framework_pkgs.zsh:22]: Can we use a variable from inside GHA instead? If they change the xcode version, this will instantly break.

Comment (erikng) [reply build_python_framework_pkgs.zsh:22]: They maintain several versions of Xcode and symbolic link to Xcode.app for their "stable paths". Even if it breaks in the future, I think we should "pin" to the one we know works until we've tested a new one.

**Comment by erikng** [comment]
> Will `--wait` _actually_ wait for it to be done? If so, that pattern is much nicer.

yep!

File: README.md [hunk 2]
```diff
@@ -71,16 +71,6 @@ Tools that should work when using the "Recommended Flavor":
...
 - [UMAD](https://github.com/macadmins/umad)
 
-## Opinionated
-This is a Python.framework that contains everything from "Recommended", and libraries that various open source projects require.
-
-This is a **kitchen sink** approach, opting for the latest known packages.
-
-Tools that should work when using the "Opinionated Flavor":
-- [Gusto's AutoPkg promotion tool](https://github.com/Gusto/it-cpe-opensource/tree/master/autopromote)
-- [Munki CloudFront Middleware](https://github.com/AaronBurchfield/CloudFront-Middleware)
```
Comment (erikng) [inline README.md:81]: I forgot about these :( I guess we can't support everything

File: build_python_framework_pkgs.zsh [hunk 3]
```diff
@@ -73,6 +77,9 @@ RP_ZIP="/tmp/relocatable-python.zip"
 MP_ZIP="/tmp/munki-pkg.zip"
 echo "Creating Python Framework - $TYPE"
 
+# Setup notary item
+$XCODE_NOTARY_PATH store-credentials --apple-id "macadmins@cleverdevops.com" --team-id "9GQZ7KUFR6" --password "$NOTARY_PASS" macadminpython
```
Comment (erikng) [inline build_python_framework_pkgs.zsh:81]: fairly certain `"$NOTARY_PASS"` is not going to exist by default unless you add an `env` property to the github action yaml file.

**Review by erikng — CHANGES_REQUESTED** [review]

**Review by erikng — APPROVED** [review]

**Pull request merged**